flake-modules/dev/list-plugins.py

```diff
@@ -129,25 +131,52 @@ def _is_excluded(path: str) -> bool:
     return True
 
 
-if __name__ == "__main__":
-    paths: list[str] = glob.glob(
-        pathname="plugins/**/*.nix",
-        recursive=True,
-    )
-
-    filtered_paths: list[str] = list(
-        filter(
-            _is_excluded,
-            paths,
-        )
-    )
+def main(args):
+    paths: list[str] = glob.glob(pathname="plugins/**/*.nix", recursive=True)
+    filtered_paths: list[str] = list(filter(_is_excluded, paths))
     filtered_paths.sort()
 
     print("| Typ | Sty | DW | path")
     print("|-----|-----|----|--------------------------------------------------------")
     for plugin_path in filtered_paths:
-        plugin: Optional[Plugin] = parse_file(
-            path=plugin_path,
-        )
+        plugin: Optional[Plugin] = parse_file(path=plugin_path)
         if plugin is not None:
-            print(plugin)
+            if (
+                (args.kind is None or plugin.kind.name.lower() == args.kind)
+                and (args.state is None or plugin.state.name.lower() == args.state)
+                and (not args.deprecation_warnings or plugin.dep_warnings)
+            ):
+                print(plugin)
+
+
+if __name__ == "__main__":
+    parser: argparse.ArgumentParser = argparse.ArgumentParser(
```

Comment:
Now that I see it, `parser: argparse.ArgumentParser = argparse.ArgumentParser` does seem a kinda redundant, I feel a little silly for suggesting it... 😕

It is probably better though. A smarter language would make this obvious using formalised constructors and type inference...

I do think `from argparse import ArgumentParser` would be better though; I think that'd get the argparse-related imports down to one line too?

Please tell me to stop commenting on the smallest unimportant details 🫠